src/back/msl/writer.rs

```diff
@@ -944,8 +953,16 @@ impl<W: Write> Writer<W> {
                     Mf::ReverseBits => "reverse_bits",
                 };
 
-                write!(self.out, "{}::{}", NAMESPACE, fun_name)?;
-                self.put_call_parameters(iter::once(arg).chain(arg1).chain(arg2), context)?;
+                if fun == Mf::Distance && scalar_argument {
+                    write!(self.out, "{}::abs(", NAMESPACE)?;
+                    self.put_expression(arg, context, true)?;
```

Comment:
I suppose these expressions should be wrapped in brackets actually

Reply:
I don't think we have any way to guarantee this, other than passing `false` here and of the other argument